packages/components/src/index.ts

```diff
@@ -38,7 +38,11 @@ export {
 	isDefinedBorder as __experimentalIsDefinedBorder,
 	isEmptyBorder as __experimentalIsEmptyBorder,
 } from './border-box-control';
-export { BorderControl as __experimentalBorderControl } from './border-control';
+export {
+	/** @deprecated Import `BorderControl` instead. */
+	BorderControl as __experimentalBorderControl,
```

Comment:
[Context](https://github.com/WordPress/gutenberg/pull/60913/files#r1738930767)

Reply:
What's the reasoning for this specific components? This and the AlignmentMatrix one?

Reply:
There is no strong reason for a consumer to switch over, and no immediate need for the library to remove the back-compat code (which is just an alias).

Reply:
> no immediate need for the library to remove the back-compat code (which is just an alias).

This is true but in my perspective, if we don't deprecate, this will never change. But if we do deprecate now, we'll be in a better position one/two years from now to actually remove the component if we wanted to.

This is why I think if we consider something deprecated, we should always add the deprecated call. It opens more doors for us in the future without any downside for the present. 

The only cases where I won't do that is if it's something that is too impactful and that is likely to trigger warnings for all users... I would add a step there: soft deprecate and communicate and add a deprecation warning a bit later. It doesn't seem to be the case for me for these components.

Reply:
This type of pros/cons weighing is already considered in the guidelines I proposed in #61099. There is a very tangible cost we shift to the consumers when we log a deprecated call. They need to hunt it down and address it, even if they get zero functional benefit in doing so. Whereas, indefinitely keeping a couple lines of back-compat code in the library is virtually free.

And when it does start to become not free for the library in aggregate, it's not too late to log a warning then. Maybe that is what you're proposing though?

Reply:
That might be the way to go. We can always add deprecation warnings in a couple of WP versions after all components have been stabilized, though I'm not sure we'll ever be able to remove the experimental exports for certain components.

Reply:
> I'm saying that it's too late at that point to add a warning because it will cause the "removal" to be delayed by a couple years at least (people to notice the new warnings and move away from it).

The thing is that we've been maintaining components for three years or so now, and we've never experienced this level of urgency to hard remove deprecated code. And for things that we do think would be better to hard remove quickly, we do in fact log deprecation warnings as soon as possible.

> On the opposite site, if you don't show the warnings, there's no incentive for people to move out. so why it's deprecated in the first place?

This is also why we wanted to define different levels of deprecation. Passive deprecations are exactly for cases where we have no reason for people to move out. It's merely an administrative change for the library. It adds clarity/simplicity for new consumers, but existing usages don't gain anything by switching over.

> The cost is very small in most cases, this is what I'm arguing about. Very few people outside Core are using these two components, so the cost is small compare to the pros of incentivizing these users to move away from these deprecated APIs.

This is about a subjective perception of cost, so it's hard to weigh objectively. At least from the usage checks we've been doing for components, [AlignmentMatrixControl](https://wpdirectory.net/search/01J85APV8KKMY2JS9NSPZK85H7) and [BorderControl](https://wpdirectory.net/search/01J85AGHXDWZS02XAYS8N2GND6) are not low-usage, relatively speaking. And we've been getting feedback from extenders that it's definitely a hassle to deal with deprecation warnings, especially in a way that works across WP versions. To me, keeping two lines of back-compat code is cheaper than the cost on the ecosystem to switch over for no tangible benefit. And whenever we start to feel the cost analysis flipping, it likely won't be too late.

I actually think it's more likely that we'll reach a point where these (stable) components need to be deprecated entirely and/or superseded by a v2, sooner than we'll reach a point where the back-compat code is too costly to maintain.

Reply:
While I stand by my reasoning. I don't mind disagreeing and committing to any approach here.

I think the important piece of feedback for me here is the latter 

> Also, I don't like that we treat the "components" package as separate from all the other APIs in Gutenberg. If we have a strategy, we should have a consistent one.

I think we should stop treating components as a special package and whatever we do for that package, we should do for the rest.

Reply:
Let's continue this discussion elsewhere - if a decision is made to apply deprecation warnings we can do so in all affected components very easily at a later moment.